contracts/swappa/RegistryHelperUniswapV2.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.8;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/uniswap/IUniswapV2Pair.sol";
+import "../interfaces/uniswap/IUniswapV2Factory.sol";
+
+contract RegistryHelperUniswapV2 {
+
+    struct PairState {
+        uint reserve0;
+        uint reserve1;
+    }
+
+    struct PairInfo {
+        IUniswapV2Pair pair;
+        address token0;
+        address token1;
+        PairState state;
+    }
+
+    function findPairs(
+        IUniswapV2Factory factory,
+        address[] calldata tokenWhitelist,
+        uint offset,
+        uint limit
```

Comment:
needed to introduce `offset` and `limit` so that the amount of work done per call is capped
otherwise, Forno complains about out of gas when trying to fetch info for > 200 pairs.

The safe limit I've found is to fetch in chunks of 100.